src/Avalonia.Controls/Control.cs

```diff
@@ -344,10 +344,10 @@ internal void ScheduleOnLoadedCore()
         /// Invoked as the first step of marking the control as loaded and raising the
         /// <see cref="Loaded"/> event.
         /// </summary>
-        internal void OnLoadedCore()
+        internal virtual void OnLoadedCore()
```

Comment:
Two things:
 1 .Should this really be virtual? I don't see it actually used anywhere. Since there were concerns about performance, and this is internal-only and not public, I would not mark it as virtual until really needed.
 2. If it is needed, OnUnloadedCore() should probably be virtual as well.

However, generally the Core methods are fixed and if you want to override something you override OnLoaded or OnUnloaded. So please let me know the use-cases here driving the change.

Reply:
You are right, I overloaded it in WindowBase class to get Loaded event work. But then decided to do it in another way.
Can be reverted now, thanks.